### PR TITLE
Cleanup error handling in solvers/miniBDD

### PR DESCRIPTION
**Pull request by xbauch**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c7404618a79ea1b8368dfe954cd6dd9fb76244cf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/84843475

File: src/solvers/miniBDD/miniBDD.cpp [hunk 1]
```diff
@@ -22,8 +22,8 @@ Author: Daniel Kroening, kroening@kroening.com
 
 void mini_bdd_nodet::remove_reference()
 {
-  // NOLINTNEXTLINE(build/deprecated)
-  assert(reference_counter!=0);
+  PRECONDITION_WITH_DIAGNOSTICS(
+    reference_counter != 0, "All references were already removed.");
```
Comment (peterschrammel) [inline src/solvers/miniBDD/miniBDD.cpp]: start with lower case, no full stop at the end (several occurrences)

**Review by peterschrammel — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c29507d628f4fe84fcc5d675b299c692d3c6c39f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85156392

**Comment by xbauch** [comment]
Lower cases and full dot fixed.

File: src/solvers/miniBDD/miniBDD.inc [hunk 2]
```diff
@@ -18,7 +18,7 @@ inline mini_bddt::mini_bddt(class mini_bdd_nodet *_node):node(_node)
 
 inline mini_bddt &mini_bddt::operator=(const mini_bddt &x)
 {
-  assert(&x!=this);
+  PRECONDITION_WITH_DIAGNOSTICS(&x!=this, "cannot assign itself");
```
Comment (tautschnig) [inline src/solvers/miniBDD/miniBDD.inc]: Nit-pick: add spacing around `!=`

File: src/solvers/miniBDD/miniBDD.inc [hunk 3]
```diff
@@ -35,45 +35,45 @@ inline mini_bddt::~mini_bddt()
...
+  PRECONDITION_WITH_DIAGNOSTICS(is_initialized(), "BDD has to be initialized");
   return node->node_number;
 }
 
 inline const mini_bddt &mini_bddt::low() const
 {
-  assert(is_initialized());
-  assert(node->node_number>=2);
+  PRECONDITION_WITH_DIAGNOSTICS(is_initialized(), "BDD has to be initialized");
+  PRECONDITION_WITH_DIAGNOSTICS(node->node_number>=2, "only non-terminal nodes have out-going edges");
```
Comment (tautschnig) [inline src/solvers/miniBDD/miniBDD.inc]: Add spacing around `>=`

File: src/solvers/miniBDD/miniBDD.inc [hunk 4]
```diff
@@ -35,45 +35,45 @@ inline mini_bddt::~mini_bddt()
...
   return node->low;
 }
 
 inline const mini_bddt &mini_bddt::high() const
 {
-  assert(is_initialized());
-  assert(node->node_number>=2);
+  PRECONDITION_WITH_DIAGNOSTICS(is_initialized(), "BDD has to be initialized");
+  PRECONDITION_WITH_DIAGNOSTICS(node->node_number>=2, "only non-terminal nodes have out-going edges");
```
Comment (tautschnig) [inline src/solvers/miniBDD/miniBDD.inc]: Spacing around `>=`

**Review by tautschnig — APPROVED** [review]
Please squash commits before merging.

**Comment by xbauch** [comment]
Added spaces around infix operators, squashed to one commit.

**Comment by tautschnig** [comment]
Please rebase now that #2986 is merged (to make CI pass).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 004a4a42fa49314abbeb7d5ec644618dc2e0feb6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85262185

**Pull request merged**